src/test/java/io/jenkins/plugins/eddsa_api/EddsaTest.java

```diff
@@ -37,26 +39,32 @@ public class EddsaTest implements Serializable {
     public static final String USER = "jenkins";
 
     @Rule
-    public transient GenericContainer agentContainer = new GenericContainer(new ImageFromDockerfile(
-                            SSH_AGENT_NAME, false)
-                    .withFileFromClasspath(
-                            SSH_AUTHORIZED_KEYS,
-                            AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_AUTHORIZED_KEYS)
-                    .withFileFromClasspath(
-                            SSH_KEY_PATH, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_KEY_PATH)
-                    .withFileFromClasspath(
-                            SSH_KEY_PUB_PATH, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_KEY_PUB_PATH)
-                    .withFileFromClasspath(
-                            SSH_SSHD_CONFIG, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_SSHD_CONFIG)
-                    .withFileFromClasspath(DOCKERFILE, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + DOCKERFILE))
-            .withExposedPorts(22);
+    public transient GenericContainer agentContainer = !DockerClientFactory.instance()
```

Comment:
noting https://github.com/testcontainers/testcontainers-java/issues/343  and https://github.com/testcontainers/testcontainers-java/pull/6445 which I never got time to finish.(which itself was lifted from the following

https://github.com/jenkinsci/active-directory-plugin/blob/21b309ee32d1f3a021e2b53d8fab00f095392b65/src/test/java/hudson/plugins/active_directory/docker/RequireDockerRule.java#L20-L36